src/goto-instrument/cover_location_blocks.cpp

```diff
@@ -0,0 +1,59 @@
+/*******************************************************************\
+
+Module: Coverage Instrumentation
+
+Author:
+
+\*******************************************************************/
+
+/// \file
+/// Location blocks detection for Coverage Instrumentation
+
+#include "cover_location_blocks.h"
+
+#include <util/message.h>
+
+cover_location_blockst::cover_location_blockst(
+  const goto_programt &_goto_program)
+{
+  forall_goto_program_instructions(it, _goto_program)
+  {
+    const auto &bytecode_index = it->source_location.get_java_bytecode_index();
+    if(!index_to_block.count(bytecode_index))
```

Comment:
Covering each bytecode is too much. Restrict to function entry, function exit, gotos and function calls.